File: 20221102/모범답안/20221102_06/battleground.js
/**
 * @param {int} n
 * @param {array} stat1
 * @param {array} stat2
 * @return {string}
 */
function solution(n, stat1, stat2) {
    const aa = [], bb = [];
    for (let i = 0; i < n; i++) aa[i] = [stat1[i], i];
    for (let i = 0; i < n; i++) bb[i] = [stat2[i], i];

    aa.sort((x, y) => y[0] - x[0]);
    bb.sort((x, y) => y[0] - x[0]);

    q = new Set();
    for (let i = 0; i < n; i++) {
        q.add(aa[i][1]);
        q.add(bb[i][1]);
        if (q.size == i + 1) {
            break;
        }
    }

    const answer = Array(n).fill(0);
    for (let e of q) {
        answer[e] = '1'
    }

    return answer.join("")
}

File: 20221102/모범답안/20221102_06/battleground.py
def solution(n, stat1, stat2):
    '''
    :param n: int
    :param stat1: list
    :param stat2: list
    :return: string
    '''
    x = [(int(i[1]), i[0]) for i in enumerate(stat1)]
    y = [(int(i[1]), i[0]) for i in enumerate(stat2)]
    x = sorted(x, reverse=True)
    y = sorted(y, reverse=True)

    q = set()
    for i in range(n):
        q.add(x[i][1])
        q.add(y[i][1])
        if len(q) == i + 1:
            break

    answer = ['0'] * n
    for i in q:
        answer[i] = '1'
        
    return ''.join(answer)
